### DISCUSSION TITLE
logging accelerations and velocity for multiple cars

### DISCUSSION THREAD

**Discussion by amezzali** [carla-simulator/carla#4931] [General]
Hi All,

I'm trying to get the acceleration and velocity profiles of multiple self-interacting cars (simulated on autopilot, for instance). As result, I would like to obtain a log file with timestamp, id_vehicle, accelerations and velocity value. However I'm struggling to find an efficient and simple solution

In my last attempt, I've started from the `generate_traffic.py` example, but I've not found an easy way to incorporate the `imu_sensor` for the npcs. Another possibility is to use the `automatic_control.py` example and create multiple vehicles, but I'm not sure it's the right way to proceed.

Do you have any idea on how can I approach the problem?

many thanks!

**Comment by SExpert12** [comment]
Get the world snapshot.
In world snapshot you can have all the actors in the simulator. You can then extract the information like velocity, acceleration, angular velocity of all the actors.
